lib/govuk_publishing_components/app_helpers/asset_helper.rb

```diff
@@ -0,0 +1,155 @@
+module GovukPublishingComponents
+  module AppHelpers
+    class AssetHelper
+      # This is a list of all the components that have stylesheets - so a new
+      # component needs to be added to this list rather than being laboriously
+      # added and removed from every single rendering application.
+      COMPONENT_CSS_PATHS = %w[
```

Comment:
This constant doesn't seem to be used anywhere?

Reply:
Ah I just saw it in the PR description. It seems surprising that an app would reach in and use this constant. I'd wonder if you can instead make it's usage better through a method, e.g. `GovukPublishingComponents.precompile_individual_component_stylesheet_assets`

You could save having to list all the components by doing something like: `Dir["#{GovukPublishingComponents::Config.gem_directory}/app/assets/stylesheets/govuk_publishing_components/components/_*.scss"]` to generate the list of files.

I also expect it'd be good to have:

```
  if defined? @individual_stylesheets
    list_of_stylesheets = @individual_stylesheets.get_stylesheets.map do | component |
      stylesheet_link_tag(component, integrity: false)
    end
    raw(list_of_stylesheets.join(""))
  end
```

as something that is handled by a method too, that'd allow cleaner app usage and allow the method to be tested.

Reply:
Thanks, we have refactored this.

We can now use the `get_component_css_paths` method from GovukPublishingComponents to get the list of stylesheets to precompile.

We've also added `render_component_stylesheets` to render the stylesheets in application.html